KeyCheckSMT/src/main/java/com/ably/kafka/connect/transform/RecordKeyCheck.java

```diff
@@ -0,0 +1,82 @@
+package com.ably.kafka.connect.transform;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.transforms.Transformation;
+import org.apache.kafka.connect.transforms.util.SimpleConfig;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+/*
+ * This transformation is used to check the validitiy of the key of the record and is strictly to use with the Ably sink connector.
+ * This transformation is to be used only when one of the configurations given contains #{key} in them.
+ **/
+public class RecordKeyCheck<R extends ConnectRecord<R>> implements Transformation<R> {
+    private final static String KEY_TOKEN = "#{key}";
+    private final static String CHANNEL_CONFIG = "channel.name";
+    private final static String MESSAGE_CONFIG = "message.name";
+    public static final ConfigDef CONFIG_DEF;
+    private String channelConfig;
+    private String messageNameConfig;
+
+    @Override
+    public R apply(R record) {
+        final byte[] key = (byte[]) record.key();
+        String keyString = null;
+        if (key != null && ByteArrayUtils.isUTF8Encoded(key)) {
```

Comment:
You're decoding `key` twice - first in this line, then using the `String` constructor as a result below. But, at the same time, if the bytes in `key` don't pass the encoding check then you discard them silently, which also seems a bit odd.

I'm a bit confused as to what the desired behaviour should be.

I would also suggest that if you need to explicitly check `key` bytes for strict UTF-8 support (rather than default replacements provided by the `String` constructor) then either refactor your `ByteArrayUtils` method to return the `String` (if UTF-8 encoded) or `null` (if not UTF-8 encoded) or perhaps the raised Exception (it not UTF-8 encoded).

Reply:
That's a good point. I have updated code here 5d039cd8f85ec7b61c657c574ca23fcd054d2694 to simplify flow and get rid of decoding twice. 

> I'm a bit confused as to what the desired behaviour should be.

Desired behaviour for this is that - when the key is provided it needs to be a string. This check exists in connector so that the record can be processed with or without a key taken into account. 
In our case, it needs to exist as a string if one of configuration is provided with provided token otherwise the exception is thrown. Thrown exception helps the record not be processed by connector. It also helps connector not to fail on runtime if the correct error tolerance is set.